### PR TITLE
op-service: jsonutil sorted-map

### PR DESCRIPTION
**Pull request by protolambda**
**Description**

This is a small util to get consistent JSON-encoding order of map-data. This is stacked between the `superchain-config` PR and the `registry-data-tool`, since the tool needs both the superchain registry types and the new JSON-util.

**Tests**

Tests that the encoding/decoding works as expected.

### CODE REVIEW

**Comment by protolambda** [comment]
Please do not merge until the base PR #6581 is complete and merged.

**Review by mslipper — APPROVED** [review]

**Comment by protolambda** [comment]
rebased + fixed lint

**Comment by protolambda** [comment]
merge queue seems stuck, I'm merging this

**Pull request merged**
